src/journal/Future.h

```diff
@@ -18,13 +19,16 @@ class FutureImpl;
 
 class Future {
 public:
-  typedef boost::intrusive_ptr<FutureImpl> FutureImplPtr;
-
-  Future() {}
-  Future(const FutureImplPtr &future_impl) : m_future_impl(future_impl) {}
-
-  inline bool is_valid() const {
-    return m_future_impl.get() != nullptr;
+  Future();
+  Future(const Future&);
+  Future& operator=(const Future&);
+  Future(Future&&);
+  Future& operator=(Future&&);
+  Future(const ceph::ref_t<FutureImpl>& future_impl);
+  ~Future();
+
+  bool is_valid() const {
+    return !!m_future_impl;
```

Comment:
or we can just cast it to bool, like
```c++
return bool(m_future_impl);
````
if we want to make this explicit?

Reply:
done